lib/grizzly/zwave/commands/indicator_report.ex

```diff
@@ -0,0 +1,94 @@
+defmodule Grizzly.ZWave.Commands.IndicatorReport do
+  @moduledoc """
+  This command is used to advertise the state of an indicator resource.
+
+  Params:
+
+    * `:indicator_id` - This field is used to specify the actual indicator resource (required for v2+)
+    * `:value` - Value of the unique indicator resource (required for v1 only)
+    * `:resources` - Indicator resources (required for v2+ only)
+
+  """
+
+  @behaviour Grizzly.ZWave.Command
+
+  alias Grizzly.ZWave.{Command, DecodeError}
+  alias Grizzly.ZWave.CommandClasses.Indicator
+
+  @type param :: {:resources, [Indicator.resource()]} | {:value, byte}
+
+  @impl true
+  @spec new([param()]) :: {:ok, Command.t()}
+  def new(params) do
+    command = %Command{
+      name: :indicator_report,
+      command_byte: 0x03,
+      command_class: Indicator,
+      params: params,
+      impl: __MODULE__
+    }
+
+    {:ok, command}
+  end
+
+  @impl true
+  @spec encode_params(Command.t()) :: binary()
+  def encode_params(command) do
+    value = Command.param(command, :value)
+
+    if value != nil do
```

Comment:
I think for `IndicatorReport` will always have a value.

Reply:
The doc says, somewhat confusingly:

```
A sending node MUST advertise the current value of the mapped Indicator ID and Property ID.
A version 2 controlling node MUST ignore the Indicator 0 Value field if other values are advertised.
```
I think you are right. I am ignoring it *all* the time.

Reply:
Yeah, I have read the docs a few times it is definitely confusing - but it seems that if the other fields are set we can ignore the value. However, when encoding params if only the value is set we use v1.

Might want to add a note about this in the moduledoc too as it is a little confusing and really explicit.

Reply:
Fixed

Reply:
Not quite...

Reply:
I think I got it right.